tests/TestCase/TestSuite/Fixture/SchemaLoaderTest.php

```diff
@@ -89,32 +90,29 @@ public function testLoadMissingFile(): void
      */
     public function testDropTruncateTables(): void
     {
-        $connection = $this->createMock(Connection::class);
-
-        $tableSchema = $this->createMock(TableSchema::class);
-        $schemaDialect = $this->createMock(SchemaDialect::class);
-        $schemaDialect
-            ->expects($this->atLeastOnce())->method('dropConstraintSql')->with($tableSchema)->willReturn(['']);
-        $schemaDialect
-            ->expects($this->atLeastOnce())->method('dropTableSql')->with($tableSchema)->willReturn(['']);
-        $schemaDialect
-            ->expects($this->atLeastOnce())->method('truncateTableSql')->with($tableSchema)->willReturn(['']);
-        $driver = $this->createMock(DriverInterface::class);
-        $driver
-            ->expects($this->atLeastOnce())->method('schemaDialect')->willReturn($schemaDialect);
-        $connection
-            ->expects($this->atLeastOnce())->method('getDriver')->willReturn($driver);
-
-        $schemaCollection = $this->createMock(Collection::class);
-        $schemaCollection
-            ->expects($this->atLeastOnce())->method('listTables')->willReturn(['schema_test']);
-        $schemaCollection
-            ->expects($this->atLeastOnce())->method('describe')->willReturn($tableSchema);
-        $connection
-            ->expects($this->atLeastOnce())->method('getSchemaCollection')->willReturn($schemaCollection);
-
-        ConnectionManager::setConfig('schema_test', $connection);
-        $this->loader->loadFiles([], 'schema_test', true, true);
+        $this->skipIf(!extension_loaded('pdo_sqlite'), 'Skipping as SQLite extension is missing');
+        $path = TMP . 'schema_loader_test.sqlite';
+        ConnectionManager::setConfig('test_schema_loader', [
+            'className' => Connection::class,
+            'driver' => Sqlite::class,
+            'database' => $path,
+        ]);
+        $loader = new SchemaLoader(['outputLevel' => ConsoleIo::QUIET]);
+
+        $schemaFile = $this->createSchemaFile('schema_loader_first');
+        $loader->loadFiles($schemaFile, 'test_schema_loader');
+        $connection = ConnectionManager::get('test_schema_loader');
+
+        $result = $connection->getSchemaCollection()->listTables();
+        $this->assertEquals(['schema_loader_first'], $result);
+
+        $schemaFile = $this->createSchemaFile('schema_loader_second');
+        $loader->loadFiles($schemaFile, 'test_schema_loader');
+
+        $result = $connection->getSchemaCollection()->listTables();
+        $this->assertEquals(['schema_loader_second'], $result);
```

Comment:
This needs to test the drop and truncate tables feature.

Reply:
Yea , it does. I missed that it was an equals check.

Reply:
I added an insert to the generated SQL file and assertions for the truncation.